server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -895,6 +900,15 @@ protected Node(
                 clusterService.getClusterSettings()
             );
 
+            List<HealthIndicatorService> serverHealthIndicatorServices = List.of(
+                new InstanceHasMasterHealthIndicatorService(clusterService)
+            );
```

Comment:
When we talked with Ievgen about this PR I suggested to do it like this, but I agree it can become cumbersome once many more indicators will be added, so +1 on moving this outside of Node.